synedrion/src/paillier/rsa.rs

```diff
@@ -247,8 +241,10 @@ impl<P: PaillierParams> PublicModulus<P> {
         NonZero::new(self.modulus.0).expect("the modulus is non-zero")
     }
 
-    pub fn modulus_bounded(&self) -> Bounded<P::Uint> {
-        Bounded::new(self.modulus.0, P::MODULUS_BITS).expect("the modulus can be bounded by 2^MODULUS_BITS")
+    pub fn modulus_signed(&self) -> PublicSigned<P::WideUint> {
+        // Have to return WideUint, since Uint::BITS == P::MODULUS_BITS, so it won't fit in a Signed<Uint>.
+        PublicSigned::new_positive(self.modulus.0.to_wide(), P::MODULUS_BITS)
+            .expect("the modulus can be bounded by 2^MODULUS_BITS")
```

Comment:
Well, there is a bit of inefficiency with storage, but it's only used in exponentiation, so it doesn't really affect performance (because the bound is the same). I had a `PublicUnsigned` type for a while, but it was only used for this value alone, so I decided to get rid of it.